explore/exploration.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+#include <stdint.h>
+#include <stdexcept>
+#include <algorithm>
+#include <numeric>
+#include <cstring>
+
+namespace exploration
+{
+  template<typename It>
+  void generate_epsilon_greedy(float epsilon, uint32_t top_action, It pdf_first, It pdf_last, std::random_access_iterator_tag pdf_tag)
+  {
+    if (pdf_last < pdf_first)
+      return;
+
+    size_t num_actions = pdf_last - pdf_first;
+    if (num_actions == 0)
+      return;
```

Comment:
The distribution needs to be created in this case.

Reply:
not sure what you mean?